src/test/scala/com/fulcrumgenomics/umi/GroupReadsByUmiTest.scala

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) 2016, Fulcrum Genomics LLC
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.fulcrumgenomics.umi
+
+import java.nio.file.Files
+
+import com.fulcrumgenomics.testing.{SamRecordSetBuilder, UnitSpec}
+import com.fulcrumgenomics.umi.GroupReadsByUmi.AdjacencyUmiAssigner
+import com.fulcrumgenomics.util.Io
+import htsjdk.samtools.SAMFileHeader.SortOrder
+import htsjdk.samtools.{SamReader, SamReaderFactory}
+
+import scala.collection.JavaConversions._
+
+
+/**
+  *
+  */
+class GroupReadsByUmiTest extends UnitSpec {
+  // Returns a List of the element 't' repeated 'n' times
+  private def n[T](t: T, n:Int =1): List[T] = List.tabulate(n)(x => t)
+
+  {
+    "IdentityUmiAssigner" should "group UMIs together that have the exact same sequence" in {
+      val assigner = new GroupReadsByUmi.IdentityUmiAssigner
+      val umis = Seq("AAAAAA", "AAAAAT", "ACGTAC", "ACGTAC", "AAAAAA")
+      assigner.assign(umis) should contain theSameElementsAs Map(
+        "AAAAAA" -> "AAAAAA",
+        "AAAAAT" -> "AAAAAT",
+        "ACGTAC" -> "ACGTAC",
+        "AAAAAA" -> "AAAAAA"
+      )
+    }
+  }
+
+  {
+    "SimpleErrorUmiAssigner" should "group UMIs together by mismatches" in {
+      val assigner = new GroupReadsByUmi.SimpleErrorUmiAssigner(1)
+      val umis = Seq("AAAAAA", "AAAATT", "AAAATA",
+                     "TGCACC", "TGCACG",
+                     "GGCGGC", "GGCGGC", "GGCGGC")
+
```

Comment:
The newlines are on purpose - each row represents a set of UMIs that are expected to be grouped together.